Lesson4.ipynb

```diff
@@ -29,6 +32,655 @@
+    "\n",
+    "By using `with` statement, you can get better syntax and exceptions handling. \n",
+    "\n",
+    "`with` statement simplifies error handling by wrapping common preparation and cleanup tasks. In addition, it will automatically close the file even if there are errors when opening it. The `with` statement also provides a way for ensuring that a clean-up is always used.\n",
+    "\n",
+    "### Print specific line\n",
+    "\n",
+    "#### enumerate() function\n",
+    "enumerate() function is a built-in function of Python. It allows us to loop over something and have an automatic index counter. (More details about enumerate you can find here: http://book.pythontips.com/en/latest/enumerate.html)\n",
+    "\n",
+    "Let's run the code block below to see the differences.\n",
+    "\n",
+    "*** 1. Without enumerate, we can print each row. ***\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### Print the data without enumerate\n",
+    "import csv\n",
+    "with open('Biggest Loser 2018.csv') as csvfile:\n",
+    "    readCSV = csv.reader(csvfile)\n",
+    "    for row in readCSV:\n",
+    "        print(row)\n",
+    "        break                                     # Add break statement here for avoid looping all the rows\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*** 2. Using enumerate, we can know the row index, and print the row only you want. ***"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### Print the data of first line\n",
+    "import csv\n",
+    "with open('Biggest Loser 2018.csv') as csvfile:\n",
+    "    readCSV = csv.reader(csvfile)\n",
+    "    for row_index, row in enumerate(readCSV):\n",
+    "        if row_index == 1:\n",
+    "            print(row)\n",
+    "            break"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we mentioned before, the index of a **list** starts from 0, this rule suits for everywhere in coding as well. \n",
+    "\n",
+    "CSV is a table, therefore, the index of column and row start from 0 as well.\n",
+    "\n",
+    "From the sample above, when row_index == 0, which means printing the column names, when row_index == 1, then it can print the first people's data in the table."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Total Steps for each member\n",
+    "\n",
+    "After the Biggest Loser event, I think all the attendees wonder how many steps they clock in total.\n",
+    "\n",
+    "It's not a hard question, let's do coding to get the answer.\n",
+    "\n",
+    "*** Steps statistics ***\n",
+    "\n",
+    "In this part, the premisses we need to know are two things.\n",
```

Comment:
"In this part, the premisses we need to know are two things." should read "In this part, we should know two things."